crates/ibc/src/core/ics02_client/client_type.rs

```diff
@@ -2,10 +2,29 @@ use crate::prelude::*;
 use core::fmt::{Display, Error as FmtError, Formatter};
 use serde_derive::{Deserialize, Serialize};
 
+#[cfg(not(feature = "codec"))]
+#[cfg(not(feature = "borsh"))]
```

Comment:
We should use [cfg_attr](https://doc.rust-lang.org/reference/conditional-compilation.html#the-cfg_attr-attribute) instead.

```rust
#[cfg_attr(feature = "scale-codec", derive(codec::Encode, codec::Decode))]
#[cfg_attr(feature = "borsh", derive(BorshSerialize, BorshDeserialize))]
#[derive(Clone, Debug, PartialEq, Eq, PartialOrd, Ord, Hash, Serialize, Deserialize)]
pub struct ClientId(pub String);
```
In a follow-up PR we should also put serde `Serialize`/`Deserialize` behind a feature flag

Reply:
nice cool